### DISCUSSION TITLE
userAgent util

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#25505] [Ideas]
Since nuxt is server side rendered, it would be really cool to include a built in helper like this [util from next js](https://nextjs.org/docs/app/api-reference/functions/userAgent)

```js
import { NextRequest, NextResponse, userAgent } from 'next/server'
 
export function middleware(request: NextRequest) {
  const url = request.nextUrl
  const { device } = userAgent(request)
// ^^^^^^^^^^^^^^^^^^^^^^^^^^
  const viewport = device.type === 'mobile' ? 'mobile' : 'desktop'
  url.searchParams.set('viewport', viewport)
  return NextResponse.rewrite(url)
}
```

**Comment by TheAlexLichter** [comment]
You can already use `getRequestHeader(event, "user-agent")` ☺️

**Reply by dbzx10299** [reply]
I know that, I was thinking of something that is a bit more like nuxt device but built in, this is a next js example
```js
const { device } = userAgent(request)
```
But I think this would be a nice thing to have built into nuxt as a util. For example this would return from the user agent string
```js
{
  device: {
    type: 'mobile',
   // ...
  }
}
```

They even have  a [docs](https://docs.uaparser.js.org/v2/intro/why-ua-parser-js.html) website using vitepress, maybe this could be built into nuxt

**Reply by TheAlexLichter** [reply]
That would be more suited as a module - https://github.com/nuxt-modules/device.
